src/ORM/Table.php

```diff
@@ -2272,7 +2272,7 @@ public function saveMany($entities, $options = [])
      */
     public function delete(EntityInterface $entity, $options = [])
     {
-        $options = new ArrayObject($options + [
+        $options = new ArrayObject((array)$options + [
```

Comment:
Should not be needed with proper doc block annotation above and is dangerous to be added as it cloaks abuse.
Better to throw exceptions IMO in future releases when the options array passed is not of type array.

Reply:
it is `arrayaccess` but phpstan can not detect

Reply:
Then it should be also added to doc block :)

Reply:
error was 
```php
  2275   Binary operation "+" between array|ArrayAccess and array('atomic' =>  
         true, 'checkRules' => true, '_primary' => true) results in an error.  
```

Reply:
That I requested to be fixed in recent version of phpstan
Sounds like a left over issue then that should be reported there :)

Reply:
The error is valid, you can't use `+` operator for an `ArrayObject` instance and an array. But I don't think `$options` argument should be accepting an `ArrayAccess` instance.

Reply:
:+1: I agree, but we can't fix this in a patch. As such we should documentt ArrayObject acceptance in doc block and keep the cast for now.
We can fix this for 3.next or 4.0 then properly.

Reply:
Yup, so we keep the casting for now and fix type in 4.0.